src/hooks/usePDFPreview.ts

```diff
@@ -67,59 +66,58 @@ export default function usePDFPreview(physicalDocument: TDocumentPage, documentP
 
   const annotationConfig = { showToolbar: false, showCommentsPanel: false, downloadWithAnnotations: true, printWithAnnotations: true };
 
-  let viewSDKClient = null;
-  let embedApi = null;
-  let annotationManagerApi = null;
+  let viewSDKClient_CACHE = null;
+  let embedApi_CACHE = null;
 
-  const createPDFClient = (startingPassage: number, onLoadCallback?: Function) => {
-    viewSDKClient = new ViewSDKClient();
-    viewSDKClient.ready().then(() => {
-      // Prevent attaching preview to non-existent div
-      const pdfDiv = document.getElementById("pdf-div");
-      if (!pdfDiv) return;
-      const previewFilePromise = viewSDKClient.previewFile(physicalDocument, adobeKey, "pdf-div", viewerConfig);
-      previewFilePromise.then((adobeViewer: any) => {
-        // PDF PREVIEW -- SHOULD BE VISIBLE NOW
-        adobeViewer.getAPIs().then((api: any) => {
-          embedApi = api;
-          // if we have a passage index, scroll to it
-          if (!isNaN(Number(startingPassage))) {
-            passageIndexChangeHandler(startingPassage);
-          }
-        });
-        adobeViewer.getAnnotationManager().then((annotationManager: any) => {
-          annotationManager.setConfig(annotationConfig);
-          annotationManagerApi = annotationManager;
-          if (documentPassageMatches.length > 0) {
-            // annotationManager.addAnnotations(generateHighlights(document, documentPassageMatches));
-            documentMatchesChangeHandler(documentPassageMatches);
-          }
-        });
-        if (onLoadCallback) {
-          onLoadCallback();
-        }
-      });
-    });
+  const createPDFClient = async () => {
+    let viewSDKClient = null;
+    if (viewSDKClient_CACHE) {
+      viewSDKClient = viewSDKClient_CACHE;
```

Comment:
suggestion: given the return type of this method - could we just memoise that? e.g.

```ts
let apisMemo: { embedApi: EmbedApi, annotationManagerApi: AnnotationManagerApi };

if (apisMemo) {
  return apisMemo
} else {
  const embedApi = await adobeViewer.getAPIs();
  const annotationManagerApi = await adobeViewer.getAnnotationManager();

  return { embedApi, annotationManagerApi }
}
```

Reply:
This hasn't worked because when we get the out of memory error the annotation refuses to work at all. So (similar to the comment below), I have decided to memoize the viewerApi **only in the context of a given annotated session** / or no annotation technically counts as a session.